src/compiler/rust.rs

```diff
@@ -215,12 +219,15 @@ fn parse_dep_info<T>(dep_info: &str, cwd: T) -> Vec<String>
 }
 
 /// Run `rustc --print file-names` to get the outputs of compilation.
-fn get_compiler_outputs<T>(creator: &T, executable: &str, arguments: &[String], cwd: &str) -> SFuture<Vec<String>>
+fn get_compiler_outputs<T>(creator: &T, executable: &str, arguments: &[String], cwd: &str,
+                           env_vars: &Vec<(String, String)>) -> SFuture<Vec<String>>
     where T: CommandCreatorSync,
 {
     let mut cmd = creator.clone().new_command_sync(executable);
     cmd.args(&arguments)
         .args(&["--print", "file-names"])
+        .env_clear()
+        .envs(env_vars.iter().map(|&(ref k, ref v)| (k, v)))
```

Comment:
This seems like it's somewhat tricky to remember (clear the env before setting more env vars). I wonder if sccache could benefit from a [`Command` look alike](https://github.com/rust-lang/cargo/blob/master/src/cargo/util/process_builder.rs#L11-L17) which allows introspection (we still need to add this to libstd...) to maybe package up arguments like the executable, cwd, env, and base arguments? 

(just a random note for a future PR/refactoring)